patroni/validator.py

```diff
@@ -176,6 +176,16 @@ def is_ipv6_address(ip: str) -> bool:
     return True
 
 
+def get_bin_name(bin_name: str) -> str:
+    """Get the value of ``postgresql.bin_name[*bin_name*]`` configuration option.
+
+    :param bin_name: a key to be retrieved from ``postgresql.bin_name`` configuration.
+
+    :returns: value of ``postgresql.bin_name[*bin_name*]``, if present, otherwise *bin_name*.
+    """
+    return schema.data.get('postgresql', {}).get('bin_name', {}).get(bin_name, bin_name)
```

Comment:
why is it different to `pgcommand()`? (not `.get('bin_name', {}) or {}`)

Reply:
I found it a bit weird too, but I did that to silence `pyright`. In `pgcommand`, if there were no `or {}`, it would raise this issue:

```
$ pyright patroni/postgresql/__init__.py
WARNING: there is a new pyright version available (v1.1.308 -> v1.1.310).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

patroni/postgresql/__init__.py
  patroni/postgresql/__init__.py:240:76 - error: "get" is not a known member of "None" (reportOptionalMemberAccess)
1 error, 0 warnings, 0 informations
```

It seems `pyright` gets a bit lost on the [get function](https://github.com/zalando/patroni/blob/master/patroni/postgresql/config.py#L1175-L1176) call of `ConfigHandler` class.

Here in `get_bin_name` it's an actual dictionary `get` call, and `pyright` runs smoothly with that.

Reply:
We do something similar in other places of `patroni.postgresql` too, e.g.: https://github.com/zalando/patroni/blob/master/patroni/postgresql/__init__.py#L149-L151.

But I'm open to suggestions if you have a better solution 🙌

Reply:
I asked because we actually were thinking you are trying to solve the problem of having empty `poatgresql.bin_name` value. 🙈 
Like:
```
postgresql:
  bin_name:
  bin_dir: /bin/dir
```
When after parsing it becomes `None`

Wdyt? Can it be an issue?

Reply:
Oh, I see! I thought you were concerned about one call having `or {}` and the other not, as `get` method was supposed to fall back to a default (`{}`) already.

Yes, I did that to deal with both existing and non-existing `postgresql.bin_name` in a single line. The idea is something like "Get the value of `postgresql.bin_name.BINARY` if set, otherwise just assume the default name `BINARY` -- the latter will be the case for at least 99% of the Patroni installations 😆 ".

Reply:
> I thought you were concerned about one call having or {} and the other not

yes, that is what I am concerned about:)

>  get method was supposed to fall back to a default ({}) already

```
% cat test.yaml 
postgresql:
  bin_name:
  bin_dir: /bin/dir
```
```
>>> with open('test.yaml') as f:
...     test = yaml.safe_load(f)
>>> test
{'postgresql': {'bin_name': None, 'bin_dir': '/bin/dir'}}
>>> test.get('postgresql', {}).get('bin_name', {}).get('bin_name', 'bin_name')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
```

I mean it is covered in `pgcommand()` but here - not

Reply:
Ohh 🤦 now I got what you mean, I think you were concerned the other way around, meaning why `pgcommand` had that, and not why `get_bin_name` was missing that.

Thanks for pointing that out, makes sense! I just submitted a commit with that fix.